src/routes/goals/handlers.js

```diff
@@ -227,3 +229,36 @@ export async function retrieveGoalByIdAndRecipient(req, res) {
     await handleErrors(req, res, error, `${logContext}:RETRIEVE_GOAL_BY_ID_AND_RECIPIENT`);
   }
 }
+
+export async function getSimilarGoalsForRecipient(req, res) {
+  const recipientId = parseInt(req.params.recipientId, DECIMAL_BASE);
+  const cluster = Object.prototype.hasOwnProperty.call(req.query, 'cluster');
+  const userId = await currentUserId(req, res);
+  const user = await userById(userId);
+
+  try {
+    const similarGoalIds = await similarGoalsForRecipient(recipientId, cluster);
+
+    const ids = Array.from(similarGoalIds.reduce((acc, resp) => {
```

Comment:
This reduce doesn't work if cluster is true, as the data comes back in a different shape.
My [PR](https://github.com/HHS/Head-Start-TTADP/pull/1814) has a refactor. 
Maybe for now we don't need the non-cluster version?

Reply:
thanks for noticing this - just merged your PR